Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,213 @@
+        this.LastProposalId = 1;
+    }
+
+    public uint CreateProposal(Address recipent, ulong amount, uint votingDuration, string description)
+    {
+        var proposal = new Proposal
+        {
+            RequestedAmount = amount,
+            Description = description,
+            Recipient = recipent,
+            Owner = Message.Sender
+        };
+
+        SetProposal(LastProposalId, proposal);
+        SetVotingDeadline(LastProposalId, checked(votingDuration + Block.Number));
+        Log(new ProposalAddedLog
+        {
+            ProposalId = LastProposalId,
+            Recipent = recipent,
+            Amount = amount,
+            Description = description
+        });
+
+        return LastProposalId++;
+    }
+
+    public void Vote(uint proposalId, bool vote)
+    {
+        Assert(IsWhitelisted(Message.Sender), "The caller is not whitelisted.");
+
+        Assert(GetVotingDeadline(proposalId) > Block.Number, "Voting is closed.");
+
+        SetVote(proposalId, Message.Sender, ToVote(vote));
+    }
+
+    private void SetVote(uint proposalId, Address address, Votes vote)
+    {
+        var currentVote = (Votes)GetVote(proposalId, address);
+
+        if (currentVote == vote)
+        {
+            return;
+        }
+
+        Unvote(proposalId, currentVote);
+        State.SetUInt32($"Vote:{proposalId}:{address}", (uint)vote);
+
+        if (vote == Votes.Yes)
+        {
+            SetYesVotes(proposalId, GetYesVotes(proposalId) + 1);
+        }
+        else
+        {
+            SetNoVotes(proposalId, GetNoVotes(proposalId) + 1);
+        }
+    }
+
+    private void Unvote(uint proposalId, Votes currentVote)
+    {
+        switch (currentVote)
+        {
+            case Votes.Yes: SetYesVotes(proposalId, GetYesVotes(proposalId) - 1); break;
+            case Votes.No: SetNoVotes(proposalId, GetNoVotes(proposalId) - 1); break;
+            case Votes.None: break;
+        }
+    }
+
+    private Votes ToVote(bool vote) => vote ? Votes.Yes : Votes.No;
+
+    public void ExecuteProposal(uint proposalId)
+    {
+        var proposal = GetProposal(proposalId);
+
+        Assert(proposal.Owner == Message.Sender, "The proposal can be executed by proposal creator.");
+
+        var yesVotes = GetYesVotes(proposalId);
+        var noVotes = GetNoVotes(proposalId);
+
+        Assert(yesVotes > noVotes, "The proposal voting is not passed.");
+        Assert(yesVotes >= MinQuorum, "Min quorum for proposal is not reached.");
+
+        Assert(!proposal.Executed, "The proposal is already executed.");
+        Assert(GetVotingDeadline(proposalId) < Block.Number, "Voting is still open for the proposal.");
+        Assert(proposal.RequestedAmount <= Balance, "Insufficient balance.");
+
+        proposal.Executed = true;
+
+        SetProposal(proposalId, proposal);
+        var result = Transfer(proposal.Recipient, proposal.RequestedAmount);
+
+        Assert(result.Success, "Transfer failed.");
+
```

Comment:
A log here of some of the executed proposal details and the amount received might be nice to have.